benchmarks/time/src/main/scala/strawman/collection/immutable/HashSetBenchmark.scala

```diff
@@ -64,14 +64,15 @@ class HashSetBenchmark {
     }
   }
 
-  @Benchmark
-  def traverse_initLast(bh: Blackhole): Unit = {
-    var ys = xs
-    while (ys.nonEmpty) {
-      bh.consume(ys.last)
-      ys = ys.init
-    }
-  }
+//  // TODO: currently disabled, since it does not finish
```

Comment:
BTW do you have any idea on why this benchmark doesn’t finish?

Reply:
Yes, I do. It's a performance issue since the code calls twice per iteration `last` (once explicitly, and one time hidden in `init`) and `last` causes an iteration through the whole collections. This performance issue is solved with the CHAMP data structures, since I implemented a backwards iterator that avoids iterating front-to-back for obtaining the last element.

Reply:
Do you guys run the benchmarks regularly? I wondered why this problem wasn't spotted earlier on.

Reply:
No, we run them occasionally, and we always select a subset of the benchmarks to run.